### ISSUE TITLE
Intercept Requests Event is Broken

### ISSUE THREAD

**Issue by Morphexe** [puppeteer/puppeteer#4122]
### Steps to reproduce

This works in 1.6.2 at least, does not work in any other version I have tried after *1.12, 1.13 etc....

**Tell us about your environment:**

* Puppeteer version: 1.12 or anything above 1.6.2 it seems
* Platform / OS version: WINDOWS 10
* URLs (if applicable): www.google.com
* Node.js version: latest

**What steps will reproduce the problem?**
   Enable Intercept Request
  Load page
 all resources are now pending.

_Please include code that reproduces the issue._

```
 const browser = await launch({
    headless: false,
    args: [
      // `--proxy-server=${ROUTER_PROXY}`,
      "--no-sandbox",
      "--disable-setuid-sandbox",
      "--disable-dev-shm-usage",
      "--disable-accelerated-2d-canvas",
      "--disable-gpu",
      "--window-size=1920x1080",
      "--enable-feature=NetworkService"
    ]
  });
```

```
private interceptRequest = async (request: puppeteer.Request) => {
    console.warn("INTERCEPTED");
    await request.continue();
    return;
}

```

```
 const page = await browser.newPage();
      page.on("request", this.interceptRequest);

      this.page = page;
      await page.setViewport({ width: 1920, height: 1080 });
      await page.setUserAgent(this.agent);
      await page.setRequestInterception(true);
      page.on("response", this.interceptResponse);
      this.logger("Request Interception On");

      await page.goto("http://www.google.com");
```

**What is the expected result?**
 For the page to load  correctly

![image](https://user-images.githubusercontent.com/17990387/53833057-9da8e980-3f7f-11e9-934e-f90a197b3cd8.png)

**What happens instead?**
Currently it loads  the page without any other asset (images, fonts )

**Comment by aslushnikov** [comment]
@Morphexe the following works just fine for me on v1.13.0:

```js
const puppeteer = require('puppeteer');

async function main() {
    const browser = await puppeteer.launch({headless: false})
    const page = await browser.newPage()
    await page.setRequestInterception(true);
    page.on('request', request => {
      console.log('INTERCEPTED: ' + request.url());
      request.continue();
    });
    await page.goto('https://google.com');
}
main()
```

**Comment by Morphexe** [comment]
> @Morphexe the following works just fine for me on v1.13.0:
> 
> ```js
> const puppeteer = require('puppeteer');
> 
> async function main() {
>     const browser = await puppeteer.launch({headless: false})
>     const page = await browser.newPage()
>     await page.setRequestInterception(true);
>     page.on('request', request => {
>       console.log('INTERCEPTED: ' + request.url());
>       request.continue();
>     });
>     await page.goto('https://google.com');
> }
> main()
> ```

This is a odd one, if I replace my class method with a lambda function it works perfectly, if I don't it fails, is there something that I might be missing on my side that might be causing this? The strangest issue, is that the current way works in 1.6.2, but moving to 1.13 makes it fail

**Comment by aslushnikov** [comment]
>  is there something that I might be missing on my side that might be causing this?

@Morphexe looks like a bug somewhere in your code. You didn't post all your code, so it's hard to say for sure. If you post the whole thing, then we might be able to help.

**Comment by Morphexe** [comment]
@aslushnikov  So I found out its a scoping issue, the "this" is being rebinded by puppeter or something else on its inner workings.
Thats why it wasn't working, although I am not sure why the requests were being hanged even with that.
But I figured it out by using lambdas, so should be fine for now.

What threw me off was that  the old version was working perfectly, I was looking at examples and it was the same thing as before, so I couldn't figure out what was running through on this.
